ci/docker/Dockerfile.build.centos7

```diff
@@ -70,6 +70,13 @@ RUN yum -y check-update || true && \
         gcc-gfortran && \
     yum clean all
 
+RUN mkdir ~/ninja && \
+    cd ~/ninja &&  \
+    wget https://github.com/ninja-build/ninja/releases/download/v1.10.1/ninja-linux.zip && \
+    unzip ninja-linux.zip -d /usr/local/bin/ && \
+    alternatives --install /usr/bin/ninja ninja /usr/local/bin/ninja 1 && \
+    ninja --version
+
```

Comment:
You're introducing a dependency on cmake 3.16 and ninja 10. Both are available on Ubuntu 20.04 but not on 18.04. I just want to make sure you are conscious of that. Did you take a look at https://github.com/scikit-build/cmake-FindFortran which Brad recommended?